### ISSUE TITLE
Redstone Module with advanced circuit

### ISSUE THREAD

**Issue by duncanwebb** [TeamPneumatic/pnc-repressurized#522]
### Minecraft Version
1.14.4

### Forge Version
28.2.4

### Mod Version
pneumaticcraft-repressurized-1.14.4-1.0.6-18

### Describe your problem, including steps to reproduce it
How is the redstone module used, in comparator mode? I was expecting it to act like a normal comparator. In this case it would give a level of 1

![redstone-module](https://user-images.githubusercontent.com/4198031/84592665-a7d6ba80-ae47-11ea-9ad5-5ac9ac265268.png)

### Any other comments?
The HUD looks a bit wide.

**Comment by MuteTiefling** [comment]
The comparator mode for Redstone Modules doesn't read off of an inventory, it's instead used to compare two signals from different channels. So, you could output a signal on Red if Red's signal is higher than Blue, for instance. 

If you want to read an inventory, you still need either a vanilla Comparator or a universal sensor set to Comparator mode. These have the benefit of being able to read from a distance, so you can hide everything away behind walls or under floors.

**Comment by desht** [comment]
Yep, what @MuteTiefling said in regards to the comparator.

Regarding the HUD, that "------ITEMS/BLOCKS-----" header shouldn't be there, and is coming from another mod - certainly not from PNC. Not sure which mod's doing that, and I'm not sure how or why it's inserting text in the stat displays...

**Comment by desht** [comment]
Here's an example of usage with a vanilla comparator:

![2020-06-15_08 24 09](https://user-images.githubusercontent.com/736831/84629139-ba9dcd80-aee1-11ea-8e1c-92872d597bac.png)

**Comment by duncanwebb** [comment]
Thank you both. Now I understand how it works. Hadn't discovered that it has a receiving and a transmitting mode which is toggled by right clicking the module.

The universal sensor is very neat. When it is connected to a regular that is providing 4.9 bar and it has a security module in it leaks air. Is this intended?

**Comment by duncanwebb** [comment]
I wrote a little script to find the offending module
``` bash
#! /bin/bash
CWD=$PWD
for f in *.jar; do
    echo "$f";
    scratch=$(mktemp -d)
    pushd "${scratch}" > /dev/null
    unzip -qq "$CWD/$f"
    find . -type f -print0 | xargs -r0 strings | grep 'ITEMS/BLOCKS'
    popd > /dev/null
    rm -r "${scratch}"
done
```
It shows this:
platforms-1.14-1.5.1.jar
    "": "-------------------------------------------------- ITEMS/BLOCKS --------------------------------------------------!",
    "": "-------------------------------------------------- ITEMS/BLOCKS --------------------------------------------------!",
    "": "-------------------------------------------------- ITEMS/BLOCKS --------------------------------------------------!",

Should I raise a ticket there?

**Comment by duncanwebb** [comment]
I disabled platforms mod and got this

![name-corrections](https://user-images.githubusercontent.com/4198031/84691830-18afcc80-af45-11ea-9b8e-6de2a6fcaa4d.png)

The message is coming from
shetiphiancore-1.14-3.6.1.jar
    "": "------------------------------------------------ NAME CORRECTIONS ------------------------------------------------!",
    "": "------------------------------------------------ NAME CORRECTIONS ------------------------------------------------!",
    "": "------------------------------------------------ NAME CORRECTIONS ------------------------------------------------!",
    "": "------------------------------------------------ NAME CORRECTIONS ------------------------------------------------!",
    "": "------------------------------------------------ NAME CORRECTIONS ------------------------------------------------!",

Looks like a problem with data being carried over from one mod to another.

**Comment by desht** [comment]
How strange. I'll need to look into that but thanks for tracking down the mods of interest.

The security upgrade will cause machines to release air at 4.9 bar (specifically: 0.1 bar below the machine's danger pressure), so if you want to keep your system pressurised that highly, you could limit to e.g. 4.85 bar with the regulator module.

**Comment by MuteTiefling** [comment]
In regards to the funky text: https://github.com/NillerMedDild/Enigmatica5/issues/306

Seems they came to the same conclusion that it was shetiphiancore.

**Comment by desht** [comment]
I think I know what's going on; clashing translation keys.  Both PNC and ShetiphianCore/Platforms are using `_comment` in the lang files (so all 3 mods are actually at fault here), and those comments are what are appearing in the stat windows.

Fortunately, this should be easily fixed, and in fact should already be fixed in the latest 1.15.2 release (1.3.0 beta).  But I can fix this for 1.14.4 too.

**Comment by desht** [comment]
The odd text bug should be fixed in the 1.0.7 release for 1.14.4 (and 1.3.0 for 1.15.2).  I'd appreciate if you could try out that release and let me know if it works for you too...

**Comment by duncanwebb** [comment]
Just tested the 1.0.7 release, unfortunately it didn't make any difference.

![blocks-items](https://user-images.githubusercontent.com/4198031/84802351-c46a2280-b000-11ea-8f1a-f358580de5c5.png)

**Comment by desht** [comment]
Well, that's disappointing :(  

Update: I've been testing more and I can't reproduce the problem with 1.0.7 and both ShetiPhianCore-1.14-3.6.2 and Platforms-1.14-1.5.2 installed.  Are you on the latest versions of those mods?  In any case, I'm certain now that is being dynamically inserted by one of these mods and can't be due to clashing translation keys anymore.  PNC is not at fault here.

![2020-06-17_11 11 34](https://user-images.githubusercontent.com/736831/84885745-5dde1680-b08b-11ea-96b8-4b7b02506f14.png)

(I am fixing the missing translation keys for `gui.tab.upgrades.volume` & `gui.tab.upgrades.security` and the "Show Range" button, though)

**Comment by duncanwebb** [comment]
Many thanks, I upgraded the two mods and the messages are fixed, yet another success!

![blocks-items-fixed](https://user-images.githubusercontent.com/4198031/84909396-ed002400-b0b5-11ea-8615-f3fcd1e34a23.png)

This is the layout where the security upgrade releases pressure. 

![security-upgrade](https://user-images.githubusercontent.com/4198031/84909538-1ae56880-b0b6-11ea-8683-42ebf163c48e.png)

What do you think about setting the pressure release of the security upgrade at 4.92? The universal sensor is a valuable item, so it is worth a security upgrade.

BTW if you prefer I can raise a ticket per item.

**Comment by desht** [comment]
Excellent.

Regarding the threshold, I suppose that could be raised... as a workaround you could add an Advanced PCB to the regulator and set its threshold to say 4.85.

Regarding tickets, I don't mind a couple of issues in one ticket but probably best to keep unrelated things in their own tickets...

**Comment by duncanwebb** [comment]
Thanks, adding an advanced PCB is a good tip.

**Comment by desht** [comment]
Fixed in 1.0.8 release
